File: solutions/day-1/python/puzzle-1.py
import os

inputPath = os.path.normpath(os.path.join(os.path.dirname(os.path.abspath(__file__)), "../input.txt"))

file = open(inputPath, 'r')
input = file.read()
file.close()

sum = 0

for index in range(0, len(input) - 2):
    if input[index] == input[index + 1]:
        sum += int(input[index])

if input[len(input) - 1] == input[0]:
    sum += int(input[0])

print(sum)

File: solutions/day-1/nodejs/puzzle-1.js
#!/usr/bin/env node
"use strict";

const fs = require('fs');
const path = require('path');
const { ReverseCaptcha } = require('./reverse-captcha');

const input = fs.readFileSync(path.resolve(__dirname, '../input.txt'), 'utf-8');
ReverseCaptcha.Process({
    input: input,
    intervalFn: (index) => index + 1,
    callback: (sum) => console.log(sum)
});